### PR TITLE
MULE-18657: Default value for primaryNodeOnly parameter is not taken into account

### PR DESCRIPTION
**Pull request by ndinu**

### CODE REVIEW

**Comment by ndinu** [comment]
--validate

File: modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/source/ExtensionSourceObjectFactory.java [hunk 1]
```diff
@@ -109,6 +117,21 @@ public ExtensionMessageSource doGetObject() {
     });
   }
 
+  private Boolean calculatePrimaryNodeOnly(ResolverSet nonCallbackParameters) {
+    if (sourceModel.runsOnPrimaryNodeOnly()) {
+      return true;
+    }
+    try {
+      return (Boolean) nonCallbackParameters.getResolvers().get(PRIMARY_NODE_ONLY_PARAMETER_NAME)
```
Comment (marianogonzalez) [inline modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/source/ExtensionSourceObjectFactory.java:125]: don't assume tha tthe resolver surely exists

Comment (ndinu) [reply modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/source/ExtensionSourceObjectFactory.java:125]: What would you say should happen if the resolver is not  there?

Comment (marianogonzalez) [reply modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/source/ExtensionSourceObjectFactory.java:125]: return false;

**Comment by ndinu** [comment]
--validate

**Comment by ndinu** [comment]
--validate

**Comment by aiannucci** [comment]
--validate

**Comment by ndinu** [comment]
--validate

**Comment by ndinu** [comment]
--validate

**Pull request merged**